app_links/android/src/main/java/com/llfbandit/app_links/AppLinksHelper.java

```diff
@@ -0,0 +1,102 @@
+package com.llfbandit.app_links;
+
+import android.content.Intent;
+import android.os.Parcel;
+import android.util.Log;
+import androidx.annotation.NonNull;
+
+public class AppLinksHelper {
+    private static final String EXTRA_DATA_NAME = "com.google.firebase.dynamiclinks.DYNAMIC_LINK_DATA";
+
+    private static final String TAG = "com.llfbandit.app_links";
+
+    public String getDeepLinkFromIntent(Intent intent) {
+        String shortLink = getShortDeepLink(intent);
+
+        if (shortLink != null) {
+            Log.d(TAG, "handleIntent: (Data) (short deep link)" + shortLink);
+            return shortLink;
+        }
+
+
+        return getUrl(intent);
+    }
+
+    private String getShortDeepLink(Intent intent) {
+        byte[] bytes = intent.getByteArrayExtra(EXTRA_DATA_NAME);
+        Parcel parcel = Parcel.obtain();
+        parcel.unmarshall(bytes, 0, bytes.length);
+        parcel.setDataPosition(0);
+
+        int end = validateObjectHeader(parcel);
+
+        while (parcel.dataPosition() < end) {
+            int header = parcel.readInt();
+            switch (getFieldId(header)) {
+                case 1:
+                case 2:
+                    String shortLink = createString(parcel, header);
+                    return shortLink;
+                default:
+                    Log.d(TAG, "Nothing to show");
+                    break;
+            }
+        }
+
+        return null;
+    }
+
+    private String getUrl(Intent intent) {
+        String action = intent.getAction();
+        String dataString = intent.getDataString();
+
+        Log.d(TAG, "handleIntent: (Action) " + action);
+        Log.d(TAG, "handleIntent: (Data) " + dataString);
+
+        return dataString;
+    }
+
+    private int readSize(Parcel parcel, int header) {
+        if ((header & 0xFFFF0000) != 0xFFFF0000)
+            return header >> 16 & 0xFFFF;
+        return parcel.readInt();
+    }
+
+    public int validateObjectHeader(@NonNull Parcel p) {
+        int var1 = p.readInt();
```

Comment:
Please, name correctly the fields:
var1 => header
var2 => size
var 3=> start
20293 => MAGIC_NUMBER